internal/pool/conn.go

```diff
@@ -64,21 +64,17 @@ func (cn *Conn) RemoteAddr() net.Addr {
 }
 
 func (cn *Conn) WithReader(ctx context.Context, timeout time.Duration, fn func(rd *proto.Reader) error) error {
-	if timeout != 0 {
-		if err := cn.netConn.SetReadDeadline(cn.deadline(ctx, timeout)); err != nil {
-			return err
-		}
+	if err := cn.netConn.SetReadDeadline(cn.deadline(ctx, timeout)); err != nil {
```

Comment:
I think this should be `timeout >= 0` so it works with PubSub

Reply:
In the `pubsub.ReceiveTimeout(ctx, timeout)` case, we don't say anything about the timeout argument, is it intended to *ignore* negative timeout? 

for example `pubsub.ReceiveTimeout(ctx, -1)` means `do not set a deadline, use the previous set deadline` ?

```go
_, _ = pubsub.ReceiveTimeout(ctx, 3*time.Second) // this call set the deadline to time.Now().Add(3*time.Second)
time.Sleep(3)
_, _ = pubsub.ReceiveTimeout(ctx, -1) // this call will timeout immediately, but it may be intended to block forever
```

Reply:
`ReceiveTimeout(0)` is the same as `Receive()` which means block forever. Negative values should probably disable `Set*Deadline` calls completely.

But I admit it is subtle.